pandas/core/generic.py

```diff
@@ -10413,6 +10413,8 @@ def pct_change(self, periods=1, fill_method="pad", limit=None, freq=None, **kwar
         else:
             data = self.fillna(method=fill_method, limit=limit, axis=axis)
 
+        if freq and to_offset(freq).isAnchored():
```

Comment:
Could just be my lack of familiarity but what does `isAnchored` do here that would need to differ from the non-anchored case?

Reply:
first, the `isAnchored` isn't documented well. you can find it [here](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.tseries.offsets.DateOffset.isAnchored.html#pandas.tseries.offsets.DateOffset.isAnchored). I just found it by digging into the code.

I got to know that pct_change didn't work when the freq is anchored so I added that logic.

Reply:
so shift already does this, but .div does not (that's why you get the error); if we *always* do an ``.asfreq()`` do tests still pass?

Reply:
Oh, then I think I have about 2 options. one is that fix `.div` or another one is making the test more meticulously. I gotta try both!

Reply:
@jreback  as you pointed out, I didn't cover all cases like follow.
![image](https://user-images.githubusercontent.com/19762154/65970134-ed617c00-e4a0-11e9-9145-bf22e05b00c4.png).

And I have made tests more exhaustive now.

![image](https://user-images.githubusercontent.com/19762154/65970344-547f3080-e4a1-11e9-95af-b5805d3cf0dc.png)

Next, I'll look into `.div` :D

Reply:
that's not what I meant. .div is not freq aware (nor should it).

pls don't post pictures, just code

Reply:
ok, I'm sorry about that. And you're right the .div is under the `_flex_method_SERIES` and I think I can't do something about it. I'll verify the `.asfreq` approach

Reply:
So I think the outstanding question here is if the `to_offset(...` piece is required - is that something you took a look at @0xF4D3C0D3 ?

Reply:
I thought that handling the anchored freq was a key though, after the several testing, I have found that the approach I took was wrong.

so I take a new approach, dropping duplicate indices before reindexing.
I made a new review at the bottom.